---
title: "Questions from reading/meeting agenda"
output: word_document 
---

#Questions from reading
1) How is a credible set different from a credible interval? Is a credible set a collection of credible intervals? 
A credible set is discrete with set points (where SNPs occur) whereas credible set has a continuous distribution.

2) Why does the lead SNP cause signal noise?
The lead SNP which has the largest p value is almost never the causal variant. By adjusting for the SNPS that are in high LD with the lead SNP 
3) Go over plots: 
```{r}
knitr::include_graphics('./conditional association analysis conditioning on the lead SNP.jpg')
```

4) Why is the Bayesian method not influenced by odds ratios and RAF/MAF differences or the power of the study (related to n)? Doesn't n always influence how much we understand LD? 
Bayesian methods incorporate the variance into the approximate bayes factor and therefore, since it is incorporated, power, variance, and any properties related to sample size are already adjusted/accounted for. 

5)Where in the Bayesian framework do we incorporate functional annotations? How are these appropriately quantified into the bayes factor or are they already accounted for in the prior (due to exisitng projects such as fGWAS)? 
The functional annotations are accounted for in the prior. 

6)Overall I am having trouble framing what the solution looks like - would like to go over this so I understand what I am aiming for in the following weeks.
Adjusting the priors systematically to improve coverage. 

7) Which subset of dataset will I be using on 1000 genomes? Do I need to satisfy any conditions or assumptions of the data for my specific analysis?
Do not need to address this yet, to be determined at a later period. 
